azure-iot-device/tests/provisioning/pipeline/test_provisioning_pipeline.py

```diff
@@ -152,8 +152,7 @@ def test_connect_calls_connect_on_provider(
             assert mock_mqtt_transport.connect.call_args[1]["password"] is not None
             assert_for_symmetric_key(mock_mqtt_transport.connect.call_args[1]["password"])
         elif params_security_clients["client_class"].__name__ == "X509SecurityClient":
-            assert mock_mqtt_transport.connect.call_args[1]["client_certificate"] is not None
-            assert_for_client_x509(mock_mqtt_transport.connect.call_args[1]["client_certificate"])
+            assert mock_mqtt_transport.connect.call_args[1]["password"] is None
```

Comment:
These changes need to be made to keep the tests running. However, the fact that they even broke is because tests are not following the preferred unit testing model. These top level pipeline interface tests were actually also testing what happened at the bottom of the pipeline. These are totally separate units.

In order to make the tests pass, I have altered these tests, however, this suggests coverage is not on par with the sister class in the iothub package.